packages/core/admin/admin/src/content-manager/components/ListViewTable/BulkActionButtons/index.js

```diff
@@ -176,19 +208,153 @@ const ConfirmDialogDeleteAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading
 
 ConfirmDialogDeleteAll.propTypes = confirmDialogsPropTypes;
 
+/* -------------------------------------------------------------------------------------------------
+ * BoldChunk
+ * -----------------------------------------------------------------------------------------------*/
+
+const BoldChunk = (chunks) => <Typography fontWeight="bold">{chunks}</Typography>;
+
+/* -------------------------------------------------------------------------------------------------
+ * SelectedEntriesTableContent
+ * -----------------------------------------------------------------------------------------------*/
+
+const SelectedEntriesTableContent = ({ mainField }) => {
+  const { formatMessage } = useIntl();
+  const { selectedEntries, setSelectedEntries, rows } = useTableContext();
+
+  console.log(rows);
+
+  // Select all entries by default
+  React.useEffect(() => {
+    setSelectedEntries(rows.map((row) => row.id));
+  }, [setSelectedEntries, rows]);
+
+  return (
+    <>
+      <Typography>
+        {formatMessage(
+          {
+            id: getTrad('containers.ListPage.selectedEntriesModal.selectedCount'),
+            defaultMessage:
+              '<b>{count}</b> {count, plural, =0 {entries} one {entry} other {entries}} selected',
+          },
+          {
+            count: selectedEntries.length,
+            b: BoldChunk,
+          }
+        )}
+      </Typography>
+      <Box marginTop={5}>
+        <Table.Content>
+          <Table.Head>
+            <Table.HeaderCheckboxCell />
+            <Table.HeaderCell fieldSchemaType="number" label="id" name="id" />
+            {mainField !== 'id' && (
+              <Table.HeaderCell fieldSchemaType="string" label="name" name="name" />
+            )}
+          </Table.Head>
+          <Tbody>
+            {rows.map((entry, index) => (
+              <Tr key={entry.id}>
+                <Body.CheckboxDataCell rowId={entry.id} index={index} />
+                <Td>
+                  <Typography>{entry.id}</Typography>
+                </Td>
+                {mainField !== 'id' && (
+                  <Td>
+                    <Typography>{entry[mainField]}</Typography>
+                  </Td>
+                )}
+              </Tr>
+            ))}
+          </Tbody>
+        </Table.Content>
+      </Box>
+    </>
+  );
+};
+
+SelectedEntriesTableContent.propTypes = {
+  mainField: PropTypes.string.isRequired,
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * SelectedEntriesModal
+ * -----------------------------------------------------------------------------------------------*/
+
+const SelectedEntriesModal = ({ isOpen, onToggle, onConfirm, mainField }) => {
```

Comment:
I'm not too keen on passing the mainField 3 levels deep :(. I think you might be able to get it from the listViewDomain selector

Reply:
You're right, I updated, we're back to 0 props on SelectedEntriesTableContent